### PR TITLE
feat: notification API

### PR DESCRIPTION
**Pull request by Maxim-Gadalov**

### CODE REVIEW

File: src/main/java/com/epam/aidial/core/service/NotificationService.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        });
+
+        return notification;
+    }
+
+    public Set<Notification> listNotification(ProxyContext context) {
+        ResourceDescription notificationResource = getNotificationResource(context, encryptionService);
+        Map<String, Notification> notifications = decodeNotifications(resourceService.getResource(notificationResource));
+
+        return new TreeSet<>(notifications.values());
```
Comment (astsiapanay) [inline src/main/java/com/epam/aidial/core/service/NotificationService.java:51]: `LinkedHashMap` preserves the insertion order. I think we can just return values.

File: src/main/java/com/epam/aidial/core/service/NotificationService.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    }
+
+    public void deleteNotification(ProxyContext context, DeleteNotificationRequest request) {
+        List<String> ids = request.ids();
+        if (ids.isEmpty()) {
+            throw new IllegalArgumentException("Notification IDs cannot be empty");
+        }
+        ResourceDescription notificationResource = getNotificationResource(context, encryptionService);
+        resourceService.computeResource(notificationResource, state -> {
```
Comment (astsiapanay) [inline src/main/java/com/epam/aidial/core/service/NotificationService.java:60]: I think it's worth deleting resource if the object is empty. That would require code changes in the method `computeResource`

File: src/main/java/com/epam/aidial/core/service/NotificationService.java (hunk 1 above)
Comment (Maxim-Gadalov) [reply src/main/java/com/epam/aidial/core/service/NotificationService.java:51]: there is no guarantee that notifications will be inserted in correct sort order

File: src/main/java/com/epam/aidial/core/service/NotificationService.java (hunk 2 above)
Comment (Maxim-Gadalov) [reply src/main/java/com/epam/aidial/core/service/NotificationService.java:60]: notifications are going to be pulled frequently, so it's better to return empty body from redis compare to blob storage look ups

File: src/main/java/com/epam/aidial/core/service/NotificationService.java (hunk 1 above)
Comment (astsiapanay) [reply src/main/java/com/epam/aidial/core/service/NotificationService.java:51]: core returns time to  create notifications to front-end. Front-end expects the notifications to be sorted by time precisely.

File: src/main/java/com/epam/aidial/core/service/NotificationService.java (hunk 2 above)
Comment (astsiapanay) [reply src/main/java/com/epam/aidial/core/service/NotificationService.java:60]: let's keep as is.

**Review by astsiapanay — APPROVED** [review]

**Comment by Maxim-Gadalov** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-core-pr-345.nightly-test.deltixhub.io
>E2E tests status: success

**Pull request merged**